ooniapi/services/ooniprobe/src/ooniprobe/utils.py

```diff
@@ -50,24 +51,26 @@ def fetch_openvpn_config() -> OpenVPNConfig:
     return OpenVPNConfig(ca=ca, cert=cert.as_text(), key=key.as_text())
 
 def fetch_openvpn_endpoints() -> List[OpenVPNEndpoint]:
-    # TODO(ain): As a first step, I'm hardcoding a single endpoint. Endpoint discovery
-    # can be done at the same time than credentials renewal, but we probably want
-    # to rotate endpoints more often, design experiments etc, with a different lifecycle
-    # than credentials. A simple implementation can be more or less straightforward,
-    # but we want to dedicate some thought to the data model for the endpoint, since
-    # there might be some extra metadata that we want to expose.
-    return [
-        OpenVPNEndpoint(
-            address="51.15.187.53:1194",
-            transport="udp",
-            protocol="openvpn"
-        ),
-        OpenVPNEndpoint(
-            address="51.15.187.53:1194",
-            transport="tcp",
-            protocol="openvpn"
-        )
-    ]
+    endpoints = []
+
+    r = httpx.get(RISEUP_ENDPOINT_URL)
+    r.raise_for_status()
+    j = r.json()
+    for ep in j["gateways"]:
+        ip = ep["ip_address"]
+        # TODO(art): do we want to store this metadata somewhere?
```

Comment:
+1, I think the metadata as key-value can work for stuff like location, country code, etc coming from the provider... https://github.com/ooni/backend/pull/832/commits/c01c57636282a734b19da89bc777f370f5376b69#diff-a4debee7a43de30b4c89a21afca958ced896a3a313ebe595529ce0aead413cd8R48

for analysis it'll be useful to know when we first saw (start measuring) an endpoint, and when we ended serving it to clients.